ui/checkboxradio.js

```diff
@@ -0,0 +1,276 @@
+/*!
+ * jQuery UI Checkboxradio @VERSION
+ * http://jqueryui.com
+ *
+ * Copyright 2014 jQuery Foundation and other contributors
+ * Released under the MIT license.
+ * http://jquery.org/license
+ *
+ * http://api.jqueryui.com/checkboxradio/
+ */
+(function( factory ) {
+	if ( typeof define === "function" && define.amd ) {
+
+		// AMD. Register as an anonymous module.
+		define([
+			"jquery",
+			"./core",
+			"./widget"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory( jQuery );
+	}
+}(function( $ ) {
+
+var formResetHandler = function() {
+		var form = $( this );
+		setTimeout(function() {
+			form.find( ".ui-checkboxradio" ).checkboxradio( "refresh" );
+		});
+	},
+	escapeId = new RegExp( /([!"#$%&'()*+,./:;<=>?@[\]^`{|}~])/g );
+
+$.widget( "ui.checkboxradio", {
+	version: "@VERSION",
+	options: {
+		disabled: null,
+		label: null,
+		icon: true,
+		classes: {
+			"ui-checkboxradio-label": "ui-corner-all",
+			"ui-checkboxradio-icon": "ui-corner-all"
+		}
+	},
+
+	_getCreateOptions: function() {
+		var disabled,
+			that = this,
+			options = this._super() || {};
+
+		this._readLabel();
+
+		this.originalLabel = "";
+		this.label.contents().not( this.element ).each( function() {
+			that.originalLabel += ( this.nodeType === 3 ) ? $( this ).text() : this.outerHTML;
```

Comment:
Ok, so this is something, I was actually hoping someone would have a better idea for. Whats going on is we need to get the original label content, BUT minus the input IF the input is inside the label.

So to do this we get the contents of the label, but filter out `this.element`, then we loop over these, and if they are a text node we get the text, If its an element we get the outerHTML. The idea is we are reconstructing the content, in string form, to set the `label` value. 

The only other way i could think to do this would to be to get the label `.html()` and then get `this.element[ 0 ].outHTML` and do `labelHTML.replace( inputHTML, "" );` maybe that would actually be better thinking about it now.